lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/PostgresKgQueryStore.scala

```diff
@@ -1,29 +1,64 @@
 package io.github.tetherlessworld.mcsapps.lib.kg.stores.postgres
 
 import com.google.inject.Inject
+import io.github.tetherlessworld.mcsapps.lib.kg.models.edge.KgEdge
 import io.github.tetherlessworld.mcsapps.lib.kg.models.node.{KgNode, KgNodeContext, KgNodeLabel, KgNodeLabelContext}
 import io.github.tetherlessworld.mcsapps.lib.kg.models.path.KgPath
 import io.github.tetherlessworld.mcsapps.lib.kg.models.search.{KgSearchFacets, KgSearchQuery, KgSearchResult, KgSearchSort}
 import io.github.tetherlessworld.mcsapps.lib.kg.models.source.KgSource
 import io.github.tetherlessworld.mcsapps.lib.kg.stores.KgQueryStore
 import javax.inject.Singleton
+import slick.jdbc.GetResult
 
 import scala.concurrent.ExecutionContext
 
 @Singleton
 final class PostgresKgQueryStore @Inject()(configProvider: PostgresStoreConfigProvider)(implicit executionContext: ExecutionContext) extends AbstractPostgresKgStore(configProvider) with KgQueryStore {
   import profile.api._
 
+  private implicit val getStringList = GetResult[List[String]] (r =>
+    r.rs.getArray(r.skip.currentPos)
+      .getArray
+      .asInstanceOf[Array[Any]]
+      .toList
+      .map(_.toString())
+  )
+
+  private implicit val getKgEdge = GetResult(r => KgEdge(
+    id = r.<<[String],
+    labels = r.<<[List[String]],
+    `object` = r.<<[String],
+    predicate = r.<<[String],
+    sentences = (r.<<[String]).split(SentencesDelimChar).toList,
+    sourceIds = r.<<[List[String]],
+    subject = r.<<[String]
+  ))
+
+  private def toKgNodeLabels(rows: Seq[(NodeLabelRow, String, NodeRow, String, String)]) =
+    rows
```

Comment:
Use of the tuple makes this hard to read, but not avoidable?

Reply:
So I use it to convert the result rows into KgNodeLabels here:
https://github.com/tetherless-world/mcs-apps/blob/1853a98fba34e70168b627a7dbc6778ff42ad1e4/lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/PostgresKgQueryStore.scala#L93-L98

I think the query must return a tuple, so if I want to use a case class, it would have to be after the query is executed.

Maybe something like this?
```scala
case class KgNodeLabelQueryResultRow(nodeLabel: NodeLabelRow, nodeLabelSourceId: String...) {
  def apply(row: Tuple[NodeLabelRow, String, NodeRow, String, String]) =
    KgNodeLabelQueryResultRow(
      nodeLabel = row._1,
      ...
    )
}
...
toKgNodeLabels(runSyncTransaction(relatedNodeLabelWithNodeSourceAction).map(KgNodeLabelQueryResultRow(_))).toList
```

Reply:
Not worth the trouble in that case.